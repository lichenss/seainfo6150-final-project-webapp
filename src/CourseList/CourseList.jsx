import React from 'react'
import CourseListItem from '../CourseListItem/CourseListItem';
import styles from './CourseList.module.css'


const CourseList = (props) => {
    let displayContent;

    if (props.courses.length) {
      displayContent = (
        <ul className={styles.container}>
          {props.courses.map((course) => (
            <CourseListItem course={course} category={props.categoryName} />
          ))}
        </ul>
      );
    } else {
      displayContent = <div>You have no data!</div>;
    }
  
    return (
      <div className={styles.section}>
        {displayContent}
      </div>
    );
  };
export default CourseList