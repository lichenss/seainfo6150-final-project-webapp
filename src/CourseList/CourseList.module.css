.section {
  background-color: rgb(229, 231, 233);
  margin-top: 70px;
}

.container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  margin-top: 70px;
}

