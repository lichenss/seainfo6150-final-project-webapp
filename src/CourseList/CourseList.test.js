import React from "react";
import CourseList from "./CourseList";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
const courses={
    "cs-0001":{
"slug": "art-0001",
"title": "Therapeutic Art Life Coach Certification (Accredited)",
"description": "Therapeutic Art: Help Clients Heal, Reduce Stress & Manifest Dreams with Mandalas, Affirmations, Vision Boards & More",
"instructor": "Janet Giessl",
"image": {
  "_url": "https://arhu.umd.edu/sites/default/files/2019-04/iStock-955549248%20composite-JC%20flat2%20opt1-min.jpg",
  "_height": "619",
  "_width": "1100"
},
"requirements":"Ability to print course material."
},
"cs-0002":{
"slug": "art-0002",
"title": "The Ultimate Drawing Course - Beginner to Advanced",
"description": "Learn the #1 most important building block of all art",
"instructor": "Joeel & Natalie Rivera",
"image": {
  "_url": "https://wordpressstorageaccount.blob.core.windows.net/wp-media/wp-content/uploads/sites/946/arts-humanities-1.jpg",
  "_height": "619",
  "_width": "1100"
},
"requirements":"Motivation to learn!"
}
}
describe("CourseList tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <CourseList categoryName='computer-science'
        courses={courses}/>
        </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });