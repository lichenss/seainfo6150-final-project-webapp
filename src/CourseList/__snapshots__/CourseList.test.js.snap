// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList tests renders correctly 1`] = `
<div>
  <div
    class="section"
  >
    <div>
      You have no data!
    </div>
  </div>
</div>
`;
